app/services/data_requests/fetch_cloudwatch_logs.rb

```diff
@@ -39,53 +42,25 @@ def query_string
       QUERY
     end
 
-    def start_queries
-      results = Concurrent::Array.new
-      errors = Concurrent::Array.new
-      thread_pool = Concurrent::FixedThreadPool.new(1, fallback_policy: :abort)
-
-      dates.each do |date|
-        thread_pool.post do
-          warn "Downloading logs for #{date}"
-          results.push(wait_for_query_result(start_query(date)))
-        rescue Aws::CloudWatchLogs::Errors::InvalidParameterException => e
-          if !e.message.match?(/End time should not be before the service was generally available/)
-            errors.push(e)
-          end
-        rescue => e
-          errors.push(e)
-        end
+    def query_ranges
+      dates.map do |date|
+        in_utc = date.in_time_zone('UTC')
+        in_utc.beginning_of_day..in_utc.end_of_day
       end
-
-      thread_pool.shutdown
-      thread_pool.wait_for_termination
-
-      if errors.any?
-        warn "#{errors.count} errors"
-        raise errors.first
-      end
-
-      results
     end
 
-    def start_query(date)
-      query_start = (date - 12.hours).to_i
-      query_end = (date + 36.hours).to_i
-
-      cloudwatch_client.start_query(
-        log_group_name: 'prod_/srv/idp/shared/log/events.log',
-        start_time: query_start,
-        end_time: query_end,
-        query_string: query_string,
-      ).query_id
+    def logger
+      @logger ||= Logger.new(STDERR, level: :warn)
     end
 
-    def wait_for_query_result(query_id)
-      loop do
-        sleep 3
-        response = cloudwatch_client.get_query_results(query_id: query_id)
-        return build_result_rows_from_aws_result(response.results) if response.status == 'Complete'
-      end
+    def start_queries
+      cloudwatch_client.fetch(
+        # NOTE :from and :to are just placeholders here since we passed :slice to the constructor
+        # Maybe we can redo the options to only need one of them?
+        from: dates.min,
+        to: dates.max,
+        query: query_string,
+      )
```

Comment:
Another idea! Would love feedback on this. It would simplify the documentation for the initializer a lot

```ruby
# manual slicing
Reporting::CloudwatchClient.new.fetch(query: '', slices: [...])

# auto slicing
Reporting::CloudwatchClient.new(slice_interval: 1.day).fetch(query: '', from: '', to: '')

# error!
Reporting::CloudwatchClient.new(slice_interval: 1.day).fetch(query: '')
# => ERROR: missing [:from, :to] or :slices to #fetch
```

Reply:
went ahead with this in b1280552c7aebe2fa2e214c681bb3659b94b05a0